app/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/reportstream/ReportStreamLabOrderSenderTest.groovy

```diff
@@ -1,14 +1,97 @@
+
+        when:
+        TestApplicationContext.getImplementation(LabOrderSender).sendRequestBody("message_1", "fake token")
+        TestApplicationContext.getImplementation(LabOrderSender).sendRequestBody("message_2", "fake token")
+
+        then:
+        2 * mockClient.post(_ as String, _ as Map<String,String>, _ as String) >> "200"
+    }
+
+    def "requestToken works"() {
+        given:
+        def expected = "rs fake token"
+        def mockAuthEngine = Mock(AuthEngine)
+        def mockClient = Mock(HttpClient)
+        TestApplicationContext.register(AuthEngine, mockAuthEngine)
+        TestApplicationContext.register(HttpClient, mockClient)
+        TestApplicationContext.register(Logger, Slf4jLogger.getLogger())
+        TestApplicationContext.register(Formatter, Jackson.getInstance())
+        TestApplicationContext.injectRegisteredImplementations()
+        when:
+        def actual = ReportStreamLabOrderSender.getInstance().requestToken()
+        then:
+        1 * mockAuthEngine.generateSenderToken(_ as String, _ as String, _ as String, _ as String, 300) >> "sender fake token"
+        1 * mockClient.requestToken(_ as String, _ as Map<String,String>, _ as String) >> """{"access_token":"rs fake token", "token_type":"bearer"}"""
+        actual == expected
+    }
+
+    def "extractToken works"() {
+        given:
+        TestApplicationContext.register(Logger,Slf4jLogger.getLogger())
+        TestApplicationContext.register(Formatter, Jackson.getInstance())
+        TestApplicationContext.injectRegisteredImplementations()
+        def reportStreamLabOrderSender = ReportStreamLabOrderSender.getInstance()
+        def expected = "IaMAfaKEt0keNN"
+        def responseBody = """{"foo":"foo value", "access_token":"IaMAfaKEt0keNN", "boo":"boo value"}"""
+        when:
+        def actual = reportStreamLabOrderSender.extractToken(responseBody)
+        then:
+        actual == expected
+    }
+
+    def "composeRequestBody works"() {
+        given:
+        def reportStreamLabOrderSender = ReportStreamLabOrderSender.getInstance()
+        def expected = "scope=flexion.*.report" +
+                "&grant_type=client_credentials" +
+                "&client_assertion_type=urn:ietf:params:oauth:client-assertion-type:jwt-bearer" +
+                "&client_assertion=rsFakeToken"
+        when:
+        def actual = reportStreamLabOrderSender.composeRequestBody("rsFakeToken")
+        then:
+        actual == expected
+    }
 
     def "send order works"() {
 
         given:
+        def expected = "rs fake token"
+        def mockAuthEngine = Mock(AuthEngine)
+        def mockClient = Mock(HttpClient)
+        def mockFhir = Mock(HapiFhir)
+        TestApplicationContext.register(AuthEngine, mockAuthEngine)
+        TestApplicationContext.register(HttpClient, mockClient)
+        TestApplicationContext.register(HapiFhir,mockFhir)
+        TestApplicationContext.register(Logger, Slf4jLogger.getLogger())
```

Comment:
The Logger is already registered by the `TestApplicationContext`.

```suggestion
```

Reply:
Good Catch, we will most likely need once we do the logging; but that can wait until then.

Reply:
Even if we added logging, the `TestApplicationContext` already register a `Logger` in its `init` method.  So we should never ever need to register a `Logger` in any test.